web/components/comment-input.tsx

```diff
@@ -0,0 +1,175 @@
+import { PaperAirplaneIcon } from '@heroicons/react/solid'
```

Comment:
I moved the CommentInput and CommentInputArea methods into their own file, and made them generic. 

CommentInputArea has no changes, but CommenInput does have a few. Noticing what changed exactly is gonna be tricky when reviewing (cause of the file move), so I made a diff-checker for you to have an easier time doing so:

https://www.diffchecker.com/7L2vgLx0